discussion_forum/api.py

```diff
@@ -0,0 +1,500 @@
+import logging
+
+from rest_framework import status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from course.models import Course
+from utils import mixins as custom_mixins
+from utils.drf_utils import (
+    IsInstructorOrTAOrStudent,
+    IsOwner,
+    StandardResultsSetPagination,
+)
+from utils.utils import check_course_registration
+
+from .models import (
+    DiscussionComment,
+    DiscussionForum,
+    DiscussionReply,
+    DiscussionThread,
+)
+from .serializers import (
+    DiscussionCommentSerializer,
+    DiscussionReplySerializer,
+    DiscussionThreadSerializer,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class DiscussionThreadViewSet(
+    viewsets.GenericViewSet,
+    custom_mixins.IsRegisteredMixins,
+):
+    """Viewset for DiscussionThread."""
+
+    queryset = DiscussionThread.objects.all()
+    serializer_class = DiscussionThreadSerializer
+    permission_classes = (IsInstructorOrTAOrStudent,)
+    pagination_class = StandardResultsSetPagination
+    filterset_fields = ("title",)
+    search_fields = ("title",)
+    ordering_fields = ("id",)
+
+    @action(detail=False, methods=["POST"])
+    def create_discussion_thread(self, request):
+        """Adds a discussion thread to the discussion_forum with primary key as id.
+
+        Args:
+            request (Request): DRF `Request` object
+
+        Returns:
+            `Response` with the created discussion thread data and
+             status HTTP_201_CREATED
+
+        Raises:
+            HTTP_400_BAD_REQUEST: Raised by `is_valid()` method of the serializer
+            HTTP_404_NOT_FOUND: Raised by `_is_registered()` method
```

Comment:
keep             HTTP_404_NOT_FOUND: Raised by `_is_registered()` method at end in order(400, 401, 402 ...)